File: 37- Longest consecutive sequence/Java/solution3.java
// Optimized brute force solution:

// Time complexity: O(n)
// Space complexity: O(n)

static int longestConsecutiveSequence(int[] arr){
  if(arr.length < 2) return arr.length;
  int maxLength = 1;
  Set<Integer> values = new HashSet<>();
  for(int element : arr) values.add(element);
  for(int element : values)
    if(values.contains(element-1)) continue;
    else{
      int right = element;
      while(values.contains(right+1)) right++;
      maxLength = Math.max(maxLength, right-element+1);
    }
  return maxLength;
}

File: 37- Longest consecutive sequence/JavaScript/solution3.js
// Optimized brute force solution:

// Time complexity: O(n)
// Space complexity: O(n)

function longestConsecutiveSequence(arr){
  if(arr.length < 2) return arr.length;
  let maxLength = 1;
  let values = new Set(arr);
  values.forEach(element => {
    if(values.has(element-1)) return;
    else{
      let right = element;
      while(values.has(right+1)) right++;
      maxLength = Math.max(maxLength, right-element+1);
    }
  });
  return maxLength;
}


